lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -1987,6 +1987,13 @@ internal class EvaluatingCompiler(
         return thunkFactory.thunkEnv(metas) { value }
     }
 
+    private fun compileTime(node: DateTimeType.Time) : ThunkEnv {
+        val (hour, minute, second, nano, precision, tz_minutes, metas) = node
+        return thunkFactory.thunkEnv(metas) {
+            valueFactory.newTime(hour, minute, second, nano, precision, tz_minutes)
+        }
+    }
```

Comment:
I would make sure that the out of range time literals are checked in `ExprValueFactory.newTime` at least since everything needs to call that to to get a time instance.

Reply:
That's right. The parser catches the out of range values. The evaluator (`java.time` library to be precise) also catches the out of range values when the instance of `LocalTime` is created.